source/includes/fundamentals/code-snippets/CRUD/compoundOperations.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func main() {
+	var uri string
+	if uri = os.Getenv("DRIVER_REF_URI"); uri == "" {
+		log.Fatal("You must set your 'MONGODB_URI' environmental variable. See\n\t https://docs.mongodb.com/drivers/go/current/usage-examples/")
+	}
+
+	client, err := mongo.Connect(context.TODO(), options.Client().ApplyURI(uri))
+
+	if err != nil {
+		panic(err)
+	}
+	defer func() {
+		if err = client.Disconnect(context.TODO()); err != nil {
+			panic(err)
+		}
+	}()
+
+	client.Database("tea").Collection("ratings").Drop(context.TODO())
+
+	// begin insertDocs
+	coll := client.Database("tea").Collection("ratings")
+	docs := []interface{}{
+		bson.D{{"type", "Masala"}, {"rating", 10}},
+		bson.D{{"type", "Assam"}, {"rating", 5}},
+		bson.D{{"type", "Oolong"}, {"rating", 7}},
+		bson.D{{"type", "Earl Grey"}, {"rating", 8}},
+		bson.D{{"type", "English Breakfast"}, {"rating", 5}},
+	}
+
+	result, insertErr := coll.InsertMany(context.TODO(), docs)
+	if insertErr != nil {
+		panic(insertErr)
+	}
+	//end insertDocs
+	fmt.Printf("Number of documents inserted: %d\n", len(result.InsertedIDs))
+
+	fmt.Println("FindOneAndDelete:")
+	//begin FindOneAndDelete
+	deleteFilter := bson.D{{"type", "Assam"}}
+
+	var deleteResult bson.D
+	deleteErr := coll.FindOneAndDelete(context.TODO(), deleteFilter).Decode(&deleteResult)
```

Comment:
Optional: Typically error values in Go are just named `err`. I'm assuming the distinct error names are used to make the error assignment look identical in each result (i.e. assign with `:=` every time instead of `=` after the first assignment). You can also accomplish that by placing each example in its own block.

Optional: The variable name `deleteResult` is a bit confusing because the type returned by `DeleteOne()` is a `DeleteResult`. Consider changing it to something like `doc` or `deletedDoc`.

Example with proposed changes:
```go
fmt.Println("FindOneAndDelete:")
{
	//begin FindOneAndDelete
	filter := bson.D{{"type", "Assam"}}

	var deletedDoc bson.D
	err := coll.FindOneAndDelete(context.TODO(), filter).Decode(&deletedDoc)
	if err != nil {
		panic(err)
	}

	fmt.Println(deletedDoc)
	//end FindOneAndDelete
}
```